CMakeLists.txt

```diff
@@ -6,32 +6,53 @@
 # granted to it by virtue of its status as an intergovernmental organisation
 # nor does it submit to any jurisdiction.
 
-cmake_minimum_required( VERSION 3.12 FATAL_ERROR )
-
-# Once we update the minimum required version to 3.20,
-# we can use the command `cmake_path` instead of `get_filename_component`.
-# For discussion, see https://github.com/ecmwf-ifs/ectrans/pull/61#discussion_r1494362092
+cmake_minimum_required( VERSION 3.18 FATAL_ERROR )
+# CMake 3.17 adds INTERFACE link options which get propagated to the link stage,
+#   even if the target is linked in privately
+# CMake 3.18 allows "LINK_LANG_AND_ID" generator expression.
 
 find_package( ecbuild 3.4 REQUIRED HINTS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/../ecbuild )
 
-project( ectrans LANGUAGES C Fortran )
+project( ectrans LANGUAGES C CXX Fortran )
 include( ectrans_macros )
 
+set(CMAKE_CXX_STANDARD 17)
+
+option(IFS_SHARED_LIBS "Build shared libraries (overrides BUILD_SHARED_LIBS)" ON)
+set(BUILD_SHARED_LIBS ${IFS_SHARED_LIBS})
```

Comment:
This should not be here. Since ecbuild 3.8.4 there is, thanks to @awnawab  the possibility to override the general `BUILD_SHARED_LIBS` with project specific `ECTRANS_BUILD_SHARED_LIBS=ON|OFF`
So I suggest adapting the ifs-bundle to use latest ecbuild, and use this variable if needed.

Reply:
Your wish is my command.